sdk/cosmos/azure-cosmos/test/test_vector_policy_async.py

```diff
@@ -141,6 +142,77 @@ async def test_fail_create_vector_indexing_policy_async(self):
+            await self.test_db.create_container(
+                id='vector_container',
+                partition_key=PartitionKey(path="/id"),
+                indexing_policy=indexing_policy,
+                vector_embedding_policy=vector_embedding_policy
+            )
+            pytest.fail("Container creation should have failed for value mismatch.")
+        except exceptions.CosmosHttpResponseError as e:
+            assert e.status_code == 400
+            assert "QuantizationByteSize value :: 0 is out of range. The allowed range is between 1 and 256." \
+                   in e.http_error_message
+
+        # Pass a vector indexing policy with wrong indexingSearchListSize value
+        indexing_policy = {
+            "vectorIndexes": [
+                {"path": "/vector2", "type": "diskANN", "vectorIndexShardKey": ["/town"], "indexingSearchListSize": 5}]
+        }
+        try:
+            await self.test_db.create_container(
+                id='vector_container',
+                partition_key=PartitionKey(path="/id"),
+                indexing_policy=indexing_policy,
+                vector_embedding_policy=vector_embedding_policy
+            )
+            pytest.fail("Container creation should have failed for value mismatch.")
+        except exceptions.CosmosHttpResponseError as e:
+            assert e.status_code == 400
+            assert "IndexingSearchListSize value :: 5 is out of range. The allowed range is between 25 and 500." \
+                   in e.http_error_message
+
+        # Pass a vector indexing policy with wrong vectorIndexShardKey value
+        indexing_policy = {
+            "vectorIndexes": [
+                {"path": "/vector2", "type": "diskANN", "vectorIndexShardKey": ["country"]}]
+        }
+        try:
+            await self.test_db.create_container(
+                id='vector_container',
+                partition_key=PartitionKey(path="/id"),
+                indexing_policy=indexing_policy,
+                vector_embedding_policy=vector_embedding_policy
+            )
+            pytest.fail("Container creation should have failed for value mismatch.")
+        except exceptions.CosmosHttpResponseError as e:
+            assert e.status_code == 400
+            assert "The Vector Indexing Policy has an invalid Shard Path: country." in e.http_error_message
+
+        # Pass a vector indexing policy with too many shard paths
+        indexing_policy = {
+            "vectorIndexes": [
+                {"path": "/vector2", "type": "diskANN", "vectorIndexShardKey": ["/country", "/city", "/zipcode"]}]
+        }
+        try:
+            await self.test_db.create_container(
+                id='vector_container',
+                partition_key=PartitionKey(path="/id"),
+                indexing_policy=indexing_policy,
+                vector_embedding_policy=vector_embedding_policy
+            )
+            pytest.fail("Container creation should have failed for value mismatch.")
```

Comment:
Can you please point me to the success tests?

Reply:
Success tests are minimal, the very first ones in the file - lines 37/38 and 72.

Reply:
A serialization test validating the expected wire contract would be great, can be a follow-up.